src/GameObject.cpp

```diff
@@ -47,52 +47,61 @@ GameObject::~GameObject() {
 
 }
 
-Eigen::Vector3f& GameObject::getPosition() {
+glm::vec3& GameObject::getPosition() {
 	return position_;
 }
 
-Eigen::Vector3f& GameObject::getScale() {
+glm::vec3& GameObject::getScale() {
 	return scale_;
 }
 
 float GameObject::getYAxisRotation() {
 	return yRotationAngle_;
 }
 
-void GameObject::setPosition(Eigen::Vector3f& newPosition) {
+void GameObject::setPosition(glm::vec3& newPosition) {
 	position_ = newPosition;
 	transform.setTranslation(position_);
 }
 
-void GameObject::setScale(Eigen::Vector3f& newScale) {
+void GameObject::setScale(glm::vec3& newScale) {
 	scale_ = newScale;
 	transform.setScale(scale_);
 }
 
 void GameObject::setYAxisRotation(float angle) {
-	static Eigen::Vector3f yAxis(0.0f, 1.0f, 0.0f);
+	static glm::vec3 yAxis(0.0f, 1.0f, 0.0f);
 
 	yRotationAngle_ = angle;
 	transform.setRotate(angle, yAxis);
 }
 
 void GameObject::calculateAndSetInitialRotation() {
 	if (velocity > 0.0f) {
-		float cosOfDir = Eigen::Vector3f(1.0f, 0.0f, 0.0f).dot(direction);
+		float cosOfDir = glm::dot(glm::vec3(-1.0f, 0.0f, 0.0f), direction);
+		float rotationAngle = (glm::acos(cosOfDir) * 180.0f / M_PI);

```

Comment:
It does? ugh, when I googled it sounded like it was in degrees, but that would make sense.

Both the direction and original vector (now the -xAxis) should already be normalized so I don't think that should be an issue but it's worth a shot!